File: frontend/src/api/operation-log.ts
import request from '@/utils/request'

export interface OperationLog {
  id: number
  userId: number
  username: string
  method: string
  path: string
  query: string
  body: string
  ip: string
  userAgent: string
  status: number
  latency: number
  errorMessage: string
  operationTime: string
  createdAt: string
  updatedAt: string
}

export interface GetOperationLogsParams {
  page?: number
  pageSize?: number
  keyword?: string
  status?: number
  method?: string
  startDate?: string
  endDate?: string
}

export interface OperationLogListResult {
  items: OperationLog[]
  total: number
}

export function getOperationLogs(params: GetOperationLogsParams) {
  return request<OperationLogListResult>({
    url: '/system/operation-logs',
    method: 'get',
    params,
  })
}

export function getOperationLog(id: number) {
  return request<OperationLog>({
    url: `/system/operation-logs/${id}`,
    method: 'get',
  })
}

File: internal/model/operation_log.go
package model

import "time"

// OperationLog 操作日志
type OperationLog struct {
	BaseTenantModel
	UserID        ID        `gorm:"index" json:"userId"`
	Username      string    `gorm:"size:64" json:"username"`
	Method        string    `gorm:"size:10" json:"method"`
	Path          string    `gorm:"size:255" json:"path"`
	Query         string    `gorm:"type:text" json:"query"`
	Body          string    `gorm:"type:text" json:"body"`
	IP            string    `gorm:"size:64" json:"ip"`
	UserAgent     string    `gorm:"size:255" json:"userAgent"`
	Status        int       `json:"status"`
	Latency       int64     `json:"latency"` // 耗时(ms)
	ErrorMessage  string    `gorm:"type:text" json:"errorMessage"`
	OperationTime *DateTime `gorm:"index;<-:create" json:"operationTime"`
}

func (o *OperationLog) SetOperationTime() {
	now := DateTime(time.Now())
	o.OperationTime = &now
}
